src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -658,21 +773,22 @@ class npc_fairbanks : public CreatureScript
             SayAshbringer = false;
         }
 
-        void MoveInLineOfSight(Unit* who) override
+        //Ready to move to SmartAI
+        void SpellHit(Unit* who, SpellInfo const* spell) override
         {
-            if (who && who->GetDistance2d(me) < 2.0f)
-                if (Player* player = who->ToPlayer())
-                    if (player->HasAura(AURA_ASHBRINGER) && !SayAshbringer)
-                    {
-                        me->SetFaction(FACTION_FRIENDLY);
-                        me->SetNpcFlag(UNIT_NPC_FLAG_GOSSIP);
-                        me->CastSpell(me, 57767, true);
-                        me->SetDisplayId(16179);
-                        me->SetFacingToObject(player);
-                        SayAshbringer = true;
-                    }
-
-            SmartAI::MoveInLineOfSight(who);
+            if (who && spell->Id == ABE_ffect_000 && !SayAshbringer)
+            {
+                me->SetFaction(FACTION_FRIENDLY);
+                me->SetFacingToObject(who);
+                //There is a delay in sniffing 1615ms
+                //The sniffer uses this spell, but without the visual effect of the spell, using spell 57767 as a visual effect instead
+                me->CastSpell(me, SPELL_TRANSFORM_GHOST);
+                me->CastSpell(me, 57767, true);
```

Comment:
This is definitely a hack. Should fix the original spell instead. (Also no magic numbers =))